controllers/idler/idler_controller_test.go

```diff
@@ -583,7 +583,195 @@ func TestEnsureIdlingFailed(t *testing.T) {
 		require.EqualError(t, err, "failed to ensure idling 'john-dev': cannot set status to fail")
 	})
 }
+func TestAppNameType(t *testing.T) {
 
+	idler := &toolchainv1alpha1.Idler{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "alex-stage",
+			Labels: map[string]string{
+				toolchainv1alpha1.SpaceLabelKey: "alex",
+			},
+		},
+		Spec: toolchainv1alpha1.IdlerSpec{TimeoutSeconds: 60},
+	}
+	namespaces := []string{"dev", "stage"}
+	usernames := []string{"alex"}
+	nsTmplSet := newNSTmplSet(test.MemberOperatorNs, "alex", "advanced", "abcde11", namespaces, usernames)
+	mur := newMUR("alex")
+	reconciler, _, _, _ := prepareReconcile(t, idler.Name, getHostCluster, idler, nsTmplSet, mur)
+	payloads := preparePayloads(t, reconciler, idler.Name, "", time.Now())
+
+	t.Run("Test AppName/AppType for 'Deployment' ", func(t *testing.T) {
+		//given
+		p := func() *corev1.Pod {
+			for _, pod := range payloads.controlledPods {
+				for _, owner := range pod.OwnerReferences {
+					if owner.Kind == "ReplicaSet" && owner.Name == fmt.Sprintf("%s-replicaset", payloads.deployment.Name) {
```

Comment:
"ReplicaSet" kind is correct here because this test verifies the nested controller use case: Deployment -> ReplicaSet -> Pod.
So the scaling down the pods with a "ReplicaSet" owner should actually scale down the Deployment parent.
I would just add a comment to that test to explain it.

Reply:
I see, thanks for the explanation